contracts/src/v0.8/automation/dev/test/AutomationRegistry2_3.t.sol

```diff
@@ -88,6 +96,179 @@ contract AutomationRegistry2_3_CheckUpkeep is AutomationRegistry2_3_SetUp {
+    //record the current address and switch to the recipient
+    address originalAddr = msg.sender;
+    changePrank(ADMIN);
+
+    //mint the nonLink to the recipient
+    nonLink.mint(recipient, amount);
+
+    //change back to the original address
+    changePrank(originalAddr);
+  }
+
+  function setConfigForWithdraw() public {
+    address module = address(new ChainModuleBase());
+    AutomationRegistryBase2_3.OnchainConfig memory cfg = AutomationRegistryBase2_3.OnchainConfig({
+      paymentPremiumPPB: 10_000,
+      flatFeeMicroLink: 40_000,
+      checkGasLimit: 5_000_000,
+      stalenessSeconds: 90_000,
+      gasCeilingMultiplier: 0,
+      minUpkeepSpend: 0,
+      maxPerformGas: 10_000_000,
+      maxCheckDataSize: 5_000,
+      maxPerformDataSize: 5_000,
+      maxRevertDataSize: 5_000,
+      fallbackGasPrice: 20_000_000_000,
+      fallbackLinkPrice: 2_000_000_000, // $20
+      fallbackNativePrice: 400_000_000_000, // $4,000
+      transcoder: 0xB1e66855FD67f6e85F0f0fA38cd6fBABdf00923c,
+      registrars: s_registrars,
+      upkeepPrivilegeManager: 0xD9c855F08A7e460691F41bBDDe6eC310bc0593D8,
+      chainModule: module,
+      reorgProtectionEnabled: true,
+      financeAdmin: FINANCE_ADMIN_ADDRESS
+    });
+    bytes memory onchainConfigBytesWithBilling = abi.encode(
+      cfg,
+      new address[](0),
+      new AutomationRegistryBase2_3.BillingConfig[](0)
+    );
+    bytes memory offchainConfigBytes = abi.encode(1234, ZERO_ADDRESS);
+
+    registryMaster.setConfig(
+      s_valid_signers,
+      s_valid_transmitters,
+      F,
+      onchainConfigBytesWithBilling,
+      OFFCHAIN_CONFIG_VERSION,
+      offchainConfigBytes
+    );
+  }
+
+  function testLinkAvailableForPaymentReturnsLinkBalance() public {
+    //simulate a deposit of link to the liquidity pool
+    mintLink(address(registryMaster), 1e10);
+
+    //check there's a balance
+    assertGt(getLinkBalance(address(registryMaster)), 0);
+
+    //check the link available for payment is the link balance
+    assertEq(registryMaster.linkAvailableForPayment(), getLinkBalance(address(registryMaster)));
+  }
+
+  function testWithdrawLinkFeesRevertsBecauseOnlyFinanceAdminAllowed() public {
+    // set config with the finance admin
+    setConfigForWithdraw();
+
+    vm.expectRevert(abi.encodeWithSelector(IAutomationRegistryMaster2_3.OnlyFinanceAdmin.selector));
+    registryMaster.withdrawLinkFees(aMockAddress, 1);
+  }
+
+  function testWithdrawLinkFeesRevertsBecauseOfInsufficientBalance() public {
+    // set config with the finance admin
+    setConfigForWithdraw();
+
+    address originalAddr = msg.sender;
+    changePrank(FINANCE_ADMIN_ADDRESS);
+
+    // try to withdraw 1 link while there is 0 balance
+    vm.expectRevert(abi.encodeWithSelector(IAutomationRegistryMaster2_3.InsufficientBalance.selector, 0, 1));
+
+    registryMaster.withdrawLinkFees(aMockAddress, 1);
+
+    //change back to the original address
+    changePrank(originalAddr);
```

Comment:
I think this pattern of pranking is a touch verbose - I think we could simplify like this...

```sol
    // set config with the finance admin
    setConfigForWithdraw();

    // try to withdraw 1 link while there is 0 balance
    vm.expectRevert(abi.encodeWithSelector(IAutomationRegistryMaster2_3.InsufficientBalance.selector, 0, 1));
    vm.prank(FINANCE_ADMIN_ADDRESS);
    registryMaster.withdrawLinkFees(aMockAddress, 1);
```

... or ...

```sol
    // set config with the finance admin
    setConfigForWithdraw();

    vm.startPrank(FINANCE_ADMIN_ADDRESS); // better if we have to make more than 1 call

    // try to withdraw 1 link while there is 0 balance
    vm.expectRevert(abi.encodeWithSelector(IAutomationRegistryMaster2_3.InsufficientBalance.selector, 0, 1));
    
    registryMaster.withdrawLinkFees(aMockAddress, 1);
    vm.stopPrank();
```

Reply:
I also wouldn't be opposed to adding a `vm.stopPrank()` at the end of `setup()` - this guarantees us a "clean state" at the beginning of every test